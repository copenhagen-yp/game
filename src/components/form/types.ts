import { ChangeEvent, FormEvent, FocusEvent } from 'react';

export type formProps = {
  handleSubmit: (e: FormEvent<EventTarget>) => void;
  handleChange: (e: ChangeEvent<HTMLInputElement>) => void;
  handleBlur: (e: FocusEvent<any>) => void;
  error: {
    [key: string]: boolean;
  };
  fields: {
    [key: string]: string;
  }[];
  title: string;
  submitButtonText: string;
};
