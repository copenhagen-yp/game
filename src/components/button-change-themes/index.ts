export * from './button-change-themes';
