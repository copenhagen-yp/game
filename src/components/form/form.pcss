.form__wrapper {
  width: 100%;
}

.form {
  width: 100%
}

.form__header {
  margin-top: 80px;
  text-align: center;
}

.form__body {
  margin-top: 31px;
  margin-bottom: 100px;
  text-align: center;
}

.form__footer {
  margin-bottom: 50px;
  text-align: center;
  color: inherit;
}