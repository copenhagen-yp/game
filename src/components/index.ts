import { Button } from './button';
import { Toast } from './toast';
import { Input } from './input';

export {
  Button,
  Toast,
  Input
};