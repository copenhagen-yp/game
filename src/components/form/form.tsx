import React, { FC } from 'react';
import cn from 'classnames';

import { formProps } from './types';
import styles from './form.pcss';
import { Button } from '../../components';

export const Form: FC<formProps> = (props) => {
  const {
    children,
    wrapperClassName,
    formClassName,
    handleSubmit,
    title,
    submitButtonText,
  } = props;

  return (
    <div className={cn(styles.form__wrapper, wrapperClassName && wrapperClassName)}>
      <form onSubmit={handleSubmit} className={cn(styles.form, formClassName && formClassName)}>
        <div className={styles.form__header}>
          <h3>{title}</h3>
        </div>
        <div className={styles.form__body}>
          {children}
        </div>
        <div className={styles.form__footer}>
          <Button>{submitButtonText}</Button>
        </div>
      </form>
    </div>
  );
};
