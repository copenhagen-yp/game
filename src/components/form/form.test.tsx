import React from 'react';
import renderer from 'react-test-renderer';
import { Form } from './form'
import { Field, Input } from './../../components';

const fields = [
  {
    type: 'text',
    name: 'login',
    label: 'Логин',
  },
  {
    type: 'password',
    name: 'password',
    label: 'Пароль',
  },
];

test('Form renders correctly', () => {
  const tree = renderer
    .create(
      <Form
        onSubmit={() => {
          console.log();
        }}
        submitButtonText='Сохранить'
        title='Авторизация'
      >
        {fields.map((field) => (
          <Field
            key={field.name}
            label={field.label}
          >
            <Input
              type={field.type}
              name={field.name}
            />
          </Field>
        ))}
      </Form>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
