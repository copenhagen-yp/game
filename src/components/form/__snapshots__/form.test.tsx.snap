// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders correctly 1`] = `
<div
  className="form__wrapper"
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <div
      className="form__header"
    >
      <h3>
        Авторизация
      </h3>
    </div>
    <div
      className="form__body"
    >
      <div
        className="field"
      >
        <label
          className="field__label"
        >
          Логин
        </label>
        <input
          className="input"
          name="login"
          type="text"
        />
      </div>
      <div
        className="field"
      >
        <label
          className="field__label"
        >
          Пароль
        </label>
        <input
          className="input"
          name="password"
          type="password"
        />
      </div>
    </div>
    <div
      className="form__footer"
    >
      <button
        className="button"
      >
        Сохранить
      </button>
    </div>
  </form>
</div>
`;
